### PR TITLE
Add more sophisticated routing

### PR DESCRIPTION
**Pull request by KadenZipfel**
Closes https://github.com/protofire/omen-exchange/issues/918

### CODE REVIEW

**Comment by pimato** [comment]
works perfect <3!

The only small thing is: Do we need the `&` at the end here: 
`/#/ending/category/Coronavirus?tag=Apple&`
?

**Comment by pimato** [comment]
@hpmaxi could you please do code review here?

**Comment by KadenZipfel** [comment]
> The only small thing is: Do we need the `&` at the end here:
> `/#/ending/category/Coronavirus?tag=Apple&`
> ?

The only work around's I can think of would likely result in bugs down the road. AFAIK this is the safest way to do it.

**Comment by KadenZipfel** [comment]
Also it's worth mentioning that one gotcha with this method is that if there is an ampersand in the market state or search query, it will break. Not sure if there's a solution to this other than not using a query in the URL.

File: app/src/components/market/sections/market_list/market_home_container.tsx [hunk 1]
```diff
@@ -166,7 +244,44 @@ const MarketHomeContainer: React.FC = () => {
...
+    if(filter.category && filter.category !== 'All') {
+      route += `/category/${filter.category}`
+    }
+
+    if(filter.state && filter.state !== 'OPEN') {
+      routeQuery += `state=${filter.state}&`
+    }
+
+    if(filter.title) {
+      routeQuery += `tag=${filter.title}&`
```
Comment (hpmaxi) [inline app/src/components/market/sections/market_list/market_home_container.tsx:280]: Maybe routeQuery could be an array first with each query, and then do routeQuery.join('&').

**Pull request merged**